oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticRequestHandler.java

```diff
@@ -573,54 +575,53 @@ private List<Query> nonFullTextConstraints(IndexPlan plan, PlanResult planResult
 
         Filter filter = plan.getFilter();
         if (!filter.matchesAllTypes()) {
-            queries.add(nodeTypeConstraints(planResult.indexingRule, filter));
+            nodeTypeConstraints(planResult.indexingRule, filter).ifPresent(queries::add);
+            //queries.add(nodeTypeConstraints(planResult.indexingRule, filter));
         }
 
-        String path = FulltextIndex.getPathRestriction(plan);
-        switch (filter.getPathRestriction()) {
-        case ALL_CHILDREN:
-            if (!"/".equals(path)) {
-                queries.add(newAncestorQuery(path));
-            }
-            break;
-        case DIRECT_CHILDREN:
-            queries.add(Query.of(q -> q.bool(b -> b.must(newAncestorQuery(path)).must(newDepthQuery(path, planResult)))));
-            break;
-        case EXACT:
-            // For transformed paths, we can only add path restriction if absolute path to
-            // property can be
-            // deduced
-            if (planResult.isPathTransformed()) {
-                String parentPathSegment = planResult.getParentPathSegment();
-                if (!any.test(PathUtils.elements(parentPathSegment), "*")) {
-                    queries.add(newPathQuery(path + parentPathSegment));
-                }
-            } else {
-                queries.add(newPathQuery(path));
-            }
-            break;
-        case PARENT:
-            if (PathUtils.denotesRoot(path)) {
-                // there's no parent of the root node
-                // we add a path that can not possibly occur because there
-                // is no way to say "match no documents" in Lucene
-                queries.add(newPathQuery("///"));
-            } else {
-                // For transformed paths, we can only add path restriction if absolute path to
-                // property can be
-                // deduced
-                if (planResult.isPathTransformed()) {
-                    String parentPathSegment = planResult.getParentPathSegment();
-                    if (!any.test(PathUtils.elements(parentPathSegment), "*")) {
-                        queries.add(newPathQuery(PathUtils.getParentPath(path) + parentPathSegment));
+        if (elasticIndexDefinition.evaluatePathRestrictions()) {
```

Comment:
Is this true by default (i.e - when the property is not present on the index definition ?)

Reply:
This is actually `false` by default.

https://jackrabbit.apache.org/oak/docs/query/lucene.html#index-definition

https://github.com/apache/jackrabbit-oak/blob/trunk/oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/IndexDefinition.java#L428

Do you see any problem with that?

Reply:
No - I think so it should be ok.